### ISSUE TITLE
required option being ignored

### ISSUE THREAD

**Issue by gdgp** [plataformatec/simple_form#1457]
I am trying to create a file input field that is required, but simple_form simply ignores/removes the option specified in the code. However, it works well for other types of fields.

So, for example this code:

`f.input :video_file, as: :file, input_html: { required: true }`

Produces the following:

`<input class="file optional" type="file" name="video[video_file]" id="video_video_file">`

Is there any actual reason for this specificity towards file or is this a bug?

**Comment by nashby** [comment]
Can you please provide a sample application that reproduces the error?

**Comment by GuriSoft** [comment]
Dumb question... Why name="video[video_file]" has square brackets? Sometimes it only takes to reformulate a naming syntax for things to work correctly... My 2 cents, hope it helps.

**Comment by gdgp** [comment]
@nashby While I was putting a simple example together to show you, it worked well. 

For clarification, I re-tested the two main ways I tried to get it working before I posted the issue, and now it worked (with the other one I had tried but didn't work either):

`<%= f.input :file, as: :file, required: true %>`

So I guess I might have had some other thing interfering with it. In any case, thank you for your availability.
